tests/conftest.py

```diff
@@ -344,24 +344,26 @@ def _check_ping_for_experiment(experiment=None):
 def fixture_telemetry_event_check(trigger_experiment_loader, selenium):
     def _telemetry_event_check(experiment=None, event=None):
         fetch_events = """
-            return Services.telemetry.snapshotEvents(Ci.nsITelemetry.DATASET_ALL_CHANNELS);
+            return Glean.nimbusEvents.enrollment.testGetValue("events");
         """
 
         with selenium.context(selenium.CONTEXT_CHROME):
-            telemetry = selenium.execute_script(fetch_events)
-            logging.info(f"Event pings: {telemetry}\n")
             control = True
-            timeout = time.time() + 30
+            timeout = time.time() + 300
 
             while control and time.time() < timeout:
-                for item in telemetry.get("parent"):
-                    if (experiment and event) in item:
-                        return True
-                else:
+                telemetry = selenium.execute_script(fetch_events)
+                logging.info(f"Event pings: {telemetry}\n")
+                try:
+                    next((item for item in telemetry if experiment in item["extra"]["experiment"]))
+                    next((item for item in telemetry if event in item["name"]))
+                except StopIteration as e:
+                    logging.info(f"ERROR IS {e}\n")
+                    time.sleep(1)
                     trigger_experiment_loader()
                     continue
-            else:
-                return False
+                else:
+                    return True
```

Comment:
Using exceptions to control normal program flow is bad practice. What about using `any` instead? Something like:

```
if any(experiment in item["extra"].get("experiment", "") for item in telemetry) and \
   any(event == item["name"] for item in telemetry):
    return True

time.sleep(1)
trigger_experiment_loader()
```